Resources/Public/JavaScript/frontend_edit.js

```diff
@@ -194,22 +255,77 @@ document.addEventListener('DOMContentLoaded', function () {
   * @param {Object} jsonResponse - The JSON response containing content element data.
   */
   const renderContentElements = (jsonResponse) => {
+    debugLog(`Starting DOM assignment for ${Object.keys(jsonResponse).length} content element(s)`);
+
+    let successfulAssignments = 0;
+    let failedAssignments = 0;
+    let translationMappings = 0;
+
     for (let uid in jsonResponse) {
       const contentElement = jsonResponse[uid];
       let element = document.querySelector(`#c${uid}`);
+      let originalUid = uid;
+      let translationInfo = null;
 
+      // Handle translation mapping
       if (contentElement.element.l10n_source) {
-        element = document.querySelector(`#c${contentElement.element.l10n_source}`);
-        if (element) {
+        let l10nElement = document.querySelector(`#c${contentElement.element.l10n_source}`);
+        if (l10nElement) {
+          translationInfo = {
+            originalUid: uid,
+            translationSourceUid: contentElement.element.l10n_source,
+            language: contentElement.element.sys_language_uid || 'default'
+          };
           uid = contentElement.element.l10n_source;
+          element = l10nElement;
+          translationMappings++;
```

Comment:
These three lines may only be executed if the condition in line 273 applies. So it happens again: `l10nElement` can be null, and then `element` is also deleted in line 280.

Reply:
Sorry, I had tomatoes on my eyes and was looking at the wrong bracket. I will test it.